### PR TITLE
Create random_code_decoupling_thm.yml

### PR DESCRIPTION
**Pull request by jnelson1**
First draft for entry on using decoupling theorem to show random codes achieve channel capacity.
-Jon Nelson

### CODE REVIEW

File: codes/qubits/random_code_decoupling_thm.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# Delete instructional comments when submitting
+
+# code id, physical, logical are all lower case
+code_id: random_code_decoupling_thm
+physical: qubits
+logical: qubits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title is singular
+name: 'Achieving Quantum Channel Capacity through Random Codes'
```
Comment (mgullans) [inline codes/qubits/random_code_decoupling_thm.yml:26]: Should be "Haar random code"

File: codes/qubits/random_code_decoupling_thm.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+protection: 'Protects against any noisy quantum channel'
```
Comment (mgullans) [inline codes/qubits/random_code_decoupling_thm.yml:35]: Make more precise.  It achieves the capacity of any noisy quantum channel.

File: codes/qubits/random_code_decoupling_thm.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
...hieve perfect transmission by proving that the encoded information is decoupled from the environment. This is a necessary and sufficient condition for successful decoding since measurements of the environment should never reveal the encoded information \cite{doi:10.1103/PhysRevA.54.2629}. Intuitively, coupling with the environment can be decreased by projecting the system onto a random codespace. The more qubits that are randomly discarded, the more the codespace is decoupled from the environment. One may ask what is the least amount of qubits that can be discarded, i.e. the largest remaining codespace, that still achieves decoupling. It can be shown through the decoupling inequality \cite{arXiv:quant-ph/0512247} that the largest possible dimension of the random codespace that achieves decoupling is exactly the coherent information of the channel. Therefore, there exists codes that can transmit information with rate equal to the coherent information. Furthermore, these codes can be constructed with high probability by randomly projecting from an n-dimensional space where n is the number of physical qubits to a k-dimensional codespace where k/n is equal to the coherent information.'
```
Comment (mgullans) [inline codes/qubits/random_code_decoupling_thm.yml:33]: Nice description.  Make sure this is all original text.

**Comment by jnelson1** [comment]
I have addressed your comments in my last commit. Let me know if you have any issues with seeing my most recent commit.

File: codes/qubits/random_code_decoupling_thm.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+#   # said in a paragraph.
+#   - 'Code used in DVDs \cite{doi:####...}, 5G, etc.'
+#   - 'Realized in trapped-ion quantum devices \cite{arXiv:####.#####}, etc.'
+
+notes:
+  - 'This construction uses Haar random unitaries to choose the codespace, but it is possible to approximate this using Clifford gates. Since Clifford gates form unitary 2-designs, this protocol shows that random stabilizer codes also achieve the coherent information.'
+
+relations:
+  parents:
+    - code_id: qubits_into_qubits
```
Comment (valbert4) [inline codes/qubits/random_code_decoupling_thm.yml:96]: Change to dynamic_gen as parent

File: codes/qubits/random_code_decoupling_thm.yml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+#    \cite{manual:{(enter citation line incl. author and year here)}}
+# External websites such as code tables, coding theory packages, github pages linked as
+#    \url{https://example.com/example}
+#    \href{https://example.com/example}{link text}
+# Internal references to codes are
+#    \hyperref[code:code_id]{link text}
+# Delete instructional comments when submitting
+
+# code id, physical, logical are all lower case
+code_id: random_code_decoupling_thm
```
Comment (valbert4) [inline codes/qubits/random_code_decoupling_thm.yml:20]: haar_random

**Pull request closed without merging**